components/CustomWatermarkPicker.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { fileToDataURL } from '../lib/util'
+
+export default class extends React.Component {
+  constructor(props) {
+    super(props)
+    this.selectImage = this.selectImage.bind(this)
+  }
+
+  selectImage(e) {
+    const file = e.target ? e.target.files[0] : e
+
+    return fileToDataURL(file).then(dataURL =>
+      this.setState(() => {
```

Comment:
Why are you calling `setState` here? You don't need to if you don't have any state.